### PR TITLE
feat: add sync capability to project module

### PR DESCRIPTION
**Pull request by Alex-Izquierdo**
**Why:**
Project module misses the sync capability. An existing project can be synced (pull changes from the repo) and this have be able to be done from the module. 
jira: https://issues.redhat.com/browse/AAP-32264

**How:**
We remove the constraint to define the url when state is present because the project might already exists, url is only required when the project needs to be created, we check it in advance to check the requirement of the url and to check if the sync operation can be performed. We add the new parameter "sync" to run the synchronization (it is an async task in the backend) if the project already exists. The client needs to be updated because the backend can also return 202 for a POST operation. 

Include:
- A new sync parameter and logic related. 
- A new generic "create" method to run a post request
- A fix of `get_exactly_one` helper that generated undesired behaviors when a name search returns results based on a partial match. 
- A minor fix by initializing `project_type` variable which could be unbound. 
- Tests added and updated for the new behavior.
- A minor fix in tests that they had defined unnecessary variables that prevented to execute the tests locally. 
- Updates the project url in tests with a public valid url.

### CODE REVIEW

File: plugins/module_utils/controller.py
Comment (jshimkus-rh) [inline plugins/module_utils/controller.py:201]: ¿If there is a json decodable response set error_msg but then replace it with a different error_msg regardless?

Comment (Alex-Izquierdo) [reply plugins/module_utils/controller.py]: good catch, thanks. Fixed

**Review by jshimkus-rh — APPROVED** [review]

**Review by Akasurde — DISMISSED** [review]

**Review by alinabuzachis — APPROVED** [review]

**Review by jshimkus-rh — APPROVED** [review]

**Comment by zkayyali812** [comment]
The environments/deployments here are new to me. Should these be passing now? 

Once the tox checks pass I will approve this PR if those are broken

**Comment by Alex-Izquierdo** [comment]
@zkayyali812 @jshimkus-rh This is the good. 
Still failing a check for coverage, I'm looking how to skip it because it is already covered by integration tests. 

@Akasurde @alinabuzachis Review again if you can please. The main change is the get_exactly_one, I have changed its logic because the current implementation didn't do what it was expected. I basically changed it to support partial matches.

**Review by zkayyali812 — APPROVED** [review]
/lgtm!

**Pull request merged**
